### PR TITLE
Setup for Intended Modal Behavior

### PR DESCRIPTION
**Pull request by devashish9**
In order for all of our modals to work correctly, I did the following: 

- Make modals close when the connect/reject/accept button is clicked
- For withdraw/reject request, delete the appropriate request from db
- For accept request, placeholder function created

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -121,9 +121,11 @@ model User {
   favorites          User[]         @relation("Favorites")
   sentRequests       Request[]      @relation("sentRequests")
   receivedRequests   Request[]      @relation("receivedRequests")
-  carpools           CarpoolGroup[] @relation("userCarpools")
+  carpool            CarpoolGroup?   @relation(fields: [carpoolId], references: [id])
+  carpoolId          String?
```
Comment (jun-w-huang) [inline prisma/schema.prisma:125]: Can you explain why this change was made? am i correct in saying that carpool will hold the IDs of all the CarpoolGroups that the User is part of, then carpoolD contains the id of the CarpoolGroup that the user actually joins?

File: src/pages/index.tsx [hunk 2]
```diff
@@ -194,6 +239,10 @@ const Home: NextPage<any> = () => {
     }
   };
 
+  useEffect(() => {
```
Comment (jun-w-huang) [inline src/pages/index.tsx:242]: isn't this useEffect actually unnecessary? Rather than using a useEffect and rather than using this function to conditionally render, it should be cleaner and maybe more performance efficient if you just include both components using a ternary operator in the return method of index.tsx. eg: instead of {mapState && user && renderSidebar()} on line 257, should be {mapState && user && (sidebarState == 'explore' ? <ExploreSidebar ...etc/> : <RequestsSidebar ...etc/>)}.

The reason for this being is that the sidebars should already appropriately rerender when recommendations, favorites, or requests change, because they are already passed in as props, and so the useEffect is not necessary. 

Actually, what would be best is if both sidebars Sidebar were merged into a single component and then the conditional rendering occurs in that component. Although this technically works, I think putting both sidebars into a single component which takes in sidebarState as a prop is the cleanest solution and easiest to maintain. but maybe this could be a separate ticket.

File: src/pages/index.tsx [hunk 3]
```diff
@@ -84,6 +68,8 @@ const Home: NextPage<any> = () => {
     },
   });
 
+  // const { mutate: createGroups } = trpc.user.groups.
```
Comment (jun-w-huang) [inline src/pages/index.tsx:71]: this was accidentally left in and should be deleted?

File: src/pages/profile.tsx [hunk 4]
```diff
@@ -373,13 +373,15 @@ const Profile: NextPage = () => {
...
                           {...register("seatAvail", { valueAsNumber: true })}
                         />
-                        <Note>
-                          Registering 0 available seats will remove you from the
-                          app&apos;s recommendaiton generation.
-                        </Note>
                       </div>
                     )}
                   </div>
+                  {watch("role") == Role.DRIVER && (
```
Comment (jun-w-huang) [inline src/pages/profile.tsx:379]: Could you explain why this has been moved outside of that parent div? they should all be conditionally rendered via the already existing watch on line 360, im confused as to why you have moved it outside. I think this is also working fine in my PR.

File: src/pages/profile.tsx [hunk 5]
```diff
@@ -454,14 +456,13 @@ const Profile: NextPage = () => {
...
+                  </Note>
                   <div className="flex flex-col space-y-2">
-                    <div className="flex flex-row items-center">
-                      <Note>
-                        If you don&apos;t have set times, communicate that on
-                        your own with potential riders/drivers. For start/end
-                        time, enter whatever best matches your work schedule.
-                      </Note>
-                    </div>
+                    <div className="flex flex-row items-center"></div>
```
Comment (jun-w-huang) [inline src/pages/profile.tsx:465]: Is there a reason why this div is here if it has nothing in it?

**Review by jun-w-huang — CHANGES_REQUESTED** [review]
i don't fully understand all the backend stuff but there are some small unnecessary bits that i think can and should be removed on frontend

File: prisma/schema.prisma (hunk 1 above)
Comment (devashish9) [reply prisma/schema.prisma:125]: A User can only be part of a single CarpoolGroup. When a User is already part of a CarpoolGroup, the fields carpool and carpoolId will exist on this person. When they are not yet part of a CarpoolGroup, these properties will not exist. Earlier, for some reason that Owen and I are not really sure, we were in fact saving an array of carpools. That was really not what we wanted to do and I think it just slipped through without any of us noticing/questioning the change.

File: src/pages/index.tsx (hunk 2 above)
Comment (devashish9) [reply src/pages/index.tsx:242]: Yes, this just got left around from when i was experimenting with some weird behavior we were seeing. I'll remove both this and the commented out code.

Comment (devashish9) [reply src/pages/index.tsx:242]: I'm also going to create a few new tickets on code consolidation on frontend so we can add your notes regarding Sidebar refactoring there.

File: src/pages/profile.tsx (hunk 4 above)
Comment (devashish9) [reply src/pages/profile.tsx:379]: Yes, before these changes, this is what this note looked like:
Screenshot 2023-07-05 at 7 05 51 AM
imo, this looks weird because the buttons and the box aren't lining up. I fixed this by moving the note out of its previous position and adding this conditional rendering statement once again outside. So that it looks like this: 
Screenshot 2023-07-05 at 7 06 11 AM

I think the second one looks a lot cleaner, but I can move it back if people think the primary one is better.

File: src/pages/profile.tsx (hunk 5 above)
Comment (devashish9) [reply src/pages/profile.tsx:465]: Yeah just missed this while moving things around, I'll remove this

File: src/pages/profile.tsx (hunk 4 above)
Comment (devashish9) [reply src/pages/profile.tsx:383]: And if there's a better way to do the same thing, just lmk, I just did it as a quick fix

Comment (jun-w-huang) [reply src/pages/profile.tsx:383]: Oh okay yeah that makes sense and looks pretty good. On my PR i was working with the positioning of the Seat Availability field and so I ended up with it looking like below. I think yours actually looks better but someone will just have to handle the small merge conflicts. Also, if we choose to keep yours, just make sure to fix the typo for 'recommendation' in the note.

Screenshot 2023-07-05 at 7 55 18 AM

File: prisma/schema.prisma (hunk 1 above)
Comment (jun-w-huang) [reply prisma/schema.prisma:125]: So is carppolId a derived attribute in that it is simply the id of the carpool that the user maybe has? If so, wouldn't it make sense to just remove the carpoolId attribute and in our code we can access it by just user.carpool.id?

Comment (devashish9) [reply prisma/schema.prisma:125]: The carpoolId field is required in the User modal to create a relation between the User and Carpool Models. We can't really do away with it.

File: src/pages/profile.tsx (hunk 4 above)
Comment (devashish9) [reply src/pages/profile.tsx:383]: Sounds good

**Comment by jun-w-huang** [comment]
will need to fix some merge conflicts in SeatAvailability and prisma schema with profile_fixes_p1 PR, but otherwise LGTM

**Review by jun-w-huang — APPROVED** [review]
